File: Array/longestConsecutive.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        unorderedMap = {}
        result = 1

        for num in nums:
            unorderedMap[num] = False
        
        for num in nums:
            if num - 1 in unorderedMap: unorderedMap[num] = True

        for num in unorderedMap:
            if unorderedMap[num] == False:
                count = 1
                val = num + 1

                while unorderedMap.get(val) == True:
                    count += 1
                    val += 1
                
                result = max(result,count)

        return result

File: Array/longestConsecutive.js
function longestConsecutive(nums) {
  const hash = {};
  let result = 0;

  for (const num of nums) {
    hash[num] = true;
  }

  for (const num of nums) {
    if (hash[num - 1]) hash[num] = false;
  }

  for (const key of Object.keys(hash)) {
    if (hash[key]) {
      let temp = 0;
      let num = Number(key);

      while (hash[num] !== undefined) {
        temp++;
        num++;
      }

      result = Math.max(result, temp);
    }
  }

  return result;
}
